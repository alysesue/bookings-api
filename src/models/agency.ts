import { BaseEntity, Column, Entity, Index, PrimaryGeneratedColumn } from "typeorm";

@Entity()
export class Agency extends BaseEntity {

	@PrimaryGeneratedColumn()
	private _id: number;

	@Column()
	@Index({unique: true})
	private _name: string;

	public set name(name: string) {
		this._name = name;
	}
}
