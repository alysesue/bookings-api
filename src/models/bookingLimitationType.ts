export enum BookingLimitationType {
	NoLimitations = 'NoLimitations',
	LimitedBookingPerDate = 'LimitedBookingPerDate',
	LimitedUpcomingBooking = 'LimitedUpcomingBooking',
}
