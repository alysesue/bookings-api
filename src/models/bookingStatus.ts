export enum BookingStatus {
	PendingApproval = 1,
	Accepted = 2,
	Closed = 3,
	Cancelled = 4,
}
