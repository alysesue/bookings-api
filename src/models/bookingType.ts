export enum BookingType {
	Created = 'Created',
	Updated = 'Updated',
	CancelledOrRejected = 'CancelledOrRejected',
}
