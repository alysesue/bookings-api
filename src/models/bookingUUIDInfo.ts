export type BookingUUIDInfo = {
	bookingUUID: string;
	bookingId: number;
	serviceId: number;
	organisationId: number;
	serviceProviderId?: number;
};
