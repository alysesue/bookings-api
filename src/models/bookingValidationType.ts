export enum BookingValidationType {
	Citizen = 'citizen',
	Admin = 'admin'
}
