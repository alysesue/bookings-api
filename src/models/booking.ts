import {BaseEntity, Column, Entity, PrimaryGeneratedColumn} from "typeorm";
import {BookingStatus} from "./bookingStatus";

@Entity()
export class Booking extends BaseEntity {
	@PrimaryGeneratedColumn()
	private _id: number;

	@Column()
	private _status: BookingStatus;

	@Column()
	private _startDateTime: Date;
	@Column()
	private _sessionDurationInMinutes: number;

	constructor(startDateTime: Date, sessionDurationInMinutes: number) {
		super();
		this._startDateTime = startDateTime;
		this._sessionDurationInMinutes = sessionDurationInMinutes;
		this._status = BookingStatus.PendingApproval;
	}

	public get id(): number {
		return this._id;
	}

	public set status(newStatus: BookingStatus) {
		this._status = newStatus;
	}

	public get status(): BookingStatus {
		return this._status;
	}

	public get startDateTime(): Date {
		return this._startDateTime;
	}

	public get sessionDurationInMinutes(): number {
		return this._sessionDurationInMinutes;
	}

	public getSessionEndTime(): Date {
		return new Date(this._startDateTime.getTime() + this._sessionDurationInMinutes * 60 * 1000);
	}
}
