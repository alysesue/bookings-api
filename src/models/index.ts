import { Booking } from "./booking";
import { BookingStatus } from "./bookingStatus";
import { Schedule } from "./schedule";
import { WeekDaySchedule } from "./weekDaySchedule";
import { Calendar } from "./calendar";
import { TimeOfDay } from './timeOfDay';
import { ServiceProvider } from "./serviceProvider";
import { BusinessValidation } from './businessValidation';
import { Timeslot } from '../models/timeslot';

export { Booking, BusinessValidation, BookingStatus, Schedule, WeekDaySchedule, Calendar, TimeOfDay, Timeslot, ServiceProvider };
