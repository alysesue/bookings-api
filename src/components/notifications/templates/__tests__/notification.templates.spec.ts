import {
	CitizenEmailTemplateBookingActionByCitizen,
	CitizenEmailTemplateBookingActionByServiceProvider,
} from '../citizen.mail';
import {
	ServiceProviderEmailTemplateBookingActionByCitizen,
	ServiceProviderEmailTemplateBookingActionByServiceProvider,
} from '../serviceProviders.mail';
import { Booking, Service } from '../../../../models';

describe('Notification templates tests', () => {
	const booking = new Booking();
	booking.startDateTime = new Date('2021-04-14T02:00:00.000Z');
	booking.endDateTime = new Date('2021-04-14T03:00:00.000Z');
	booking.service = { name: 'Career' } as Service;
	booking.status = 1;
	booking.location = 'Some street';
	booking.serviceProviderId = 1;

	it('should create citizen email for citizen created booking', () => {
		const result = new CitizenEmailTemplateBookingActionByCitizen().CreatedBookingEmail(booking);
		expect(result).toMatchSnapshot();
	});

	it('should create citizen email for citizen updated booking', () => {
		const result = new CitizenEmailTemplateBookingActionByCitizen().UpdatedBookingEmail(booking);
		expect(result).toMatchSnapshot();
	});

	it('should create citizen email for citizen cancelled booking', () => {
		const result = new CitizenEmailTemplateBookingActionByCitizen().CancelledBookingEmail(booking);
		expect(result).toMatchSnapshot();
	});

	it('should create citizen email for service provider created booking', () => {
		const result = new CitizenEmailTemplateBookingActionByServiceProvider().CreatedBookingEmail(booking);
		expect(result).toMatchSnapshot();
	});

	it('should create citizen email for service provider updated booking', () => {
		const result = new CitizenEmailTemplateBookingActionByServiceProvider().UpdatedBookingEmail(booking);
		expect(result).toMatchSnapshot();
	});

	it('should create citizen email for service provider cancelled booking', () => {
		const result = new CitizenEmailTemplateBookingActionByServiceProvider().CancelledBookingEmail(booking);
		expect(result).toMatchSnapshot();
	});

	it('should create service provider email for citizen created booking', () => {
		const result = new ServiceProviderEmailTemplateBookingActionByCitizen().CreatedBookingEmail(booking);
		expect(result).toMatchSnapshot();
	});

	it('should create service provider email for citizen updated booking', () => {
		const result = new ServiceProviderEmailTemplateBookingActionByCitizen().UpdatedBookingEmail(booking);
		expect(result).toMatchSnapshot();
	});

	it('should create service provider email for citizen cancelled booking', () => {
		const result = new ServiceProviderEmailTemplateBookingActionByCitizen().CancelledBookingEmail(booking);
		expect(result).toMatchSnapshot();
	});

	it('should create service provider email for service provider updated booking', () => {
		const result = new ServiceProviderEmailTemplateBookingActionByServiceProvider().UpdatedBookingEmail(booking);
		expect(result).toMatchSnapshot();
	});

	it('should create service provider email for service provider cancelled booking', () => {
		const result = new ServiceProviderEmailTemplateBookingActionByServiceProvider().CancelledBookingEmail(booking);
		expect(result).toMatchSnapshot();
	});
});
