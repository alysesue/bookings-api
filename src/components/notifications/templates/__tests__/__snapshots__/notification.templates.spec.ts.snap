// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification templates tests should create citizen email for citizen cancelled booking 1`] = `
Object {
  "html": "<pre>
You have cancelled the following booking.
<br />
Booking for: <b>Career.</b>
<br />
Booking status: <b>Pending Approval</b>
Date: <b>14 April 2021</b>
Time: <b>10:00am - 11:00am</b>
Video Conference Link: <a href='http://www.zoom.us/1234567'>http://www.zoom.us/1234567</a>
Location: <b>Some street</b>
</pre>",
  "subject": "BookingSG cancellation: Career",
}
`;

exports[`Notification templates tests should create citizen email for citizen created booking 1`] = `
Object {
  "html": "<pre>
Your booking request has been received.
<br />
Booking for: <b>Career.</b>
<br />
Below is a confirmation of your booking details.
Booking status: <b>Pending Approval</b>
Date: <b>14 April 2021</b>
Time: <b>10:00am - 11:00am</b>
Video Conference Link: <a href='http://www.zoom.us/1234567'>http://www.zoom.us/1234567</a>
Location: <b>Some street</b>
</pre>",
  "subject": "BookingSG confirmation: Career",
}
`;

exports[`Notification templates tests should create citizen email for citizen updated booking 1`] = `
Object {
  "html": "<pre>
You have updated a booking.
<br />
Booking for: <b>Career.</b>
<br />
Below is a confirmation of your updated booking details.
Booking status: <b>Pending Approval</b>
Date: <b>14 April 2021</b>
Time: <b>10:00am - 11:00am</b>
Video Conference Link: <a href='http://www.zoom.us/1234567'>http://www.zoom.us/1234567</a>
Location: <b>Some street</b>
</pre>",
  "subject": "BookingSG update: Career",
}
`;

exports[`Notification templates tests should create citizen email for service provider cancelled booking 1`] = `
Object {
  "html": "<pre>
The following booking has been cancelled by the other party.

<br />
Booking for: <b>Career.</b>
<br />
Booking status: <b>Pending Approval</b>
Date: <b>14 April 2021</b>
Time: <b>10:00am - 11:00am</b>
Video Conference Link: <a href='http://www.zoom.us/1234567'>http://www.zoom.us/1234567</a>
Location: <b>Some street</b>
</pre>",
  "subject": "BookingSG cancellation: Career",
}
`;

exports[`Notification templates tests should create citizen email for service provider created booking 1`] = `
Object {
  "html": "<pre>
A booking has been made.
<br />
Booking for: <b>Career.</b>
<br />
Below is a confirmation of your booking details.
Booking status: <b>Pending Approval</b>
Date: <b>14 April 2021</b>
Time: <b>10:00am - 11:00am</b>
Video Conference Link: <a href='http://www.zoom.us/1234567'>http://www.zoom.us/1234567</a>
Location: <b>Some street</b>
</pre>",
  "subject": "BookingSG confirmation: Career",
}
`;

exports[`Notification templates tests should create citizen email for service provider updated booking 1`] = `
Object {
  "html": "<pre>
There has been an update to your booking confirmation.
<br />
Booking for: <b>Career.</b>
<br />
Below is a confirmation of your updated booking details.
Booking status: <b>Pending Approval</b>
Date: <b>14 April 2021</b>
Time: <b>10:00am - 11:00am</b>
Video Conference Link: <a href='http://www.zoom.us/1234567'>http://www.zoom.us/1234567</a>
Location: <b>Some street</b>
</pre>",
  "subject": "BookingSG update: Career",
}
`;

exports[`Notification templates tests should create service provider email for citizen cancelled booking 1`] = `
Object {
  "html": "<pre>
The following booking has been cancelled by the other party.
<br />
Booking for: <b>Career.</b>
<br />
Booking status: <b>Pending Approval</b>
Date: <b>14 April 2021</b>
Time: <b>10:00am - 11:00am</b>
Video Conference Link: <a href='http://www.zoom.us/1234567'>http://www.zoom.us/1234567</a>
Location: <b>Some street</b>
</pre>",
  "subject": "BookingSG cancellation: Career",
}
`;

exports[`Notification templates tests should create service provider email for citizen created booking 1`] = `
Object {
  "html": "<pre>
You have received a new booking request.
<br />
Booking for: <b>Career.</b>
<br />
Below is a summary of the booking request details.
<br/>
Booking status: <b>Pending Approval</b>
Date: <b>14 April 2021</b>
Time: <b>10:00am - 11:00am</b>
Video Conference Link: <a href='http://www.zoom.us/1234567'>http://www.zoom.us/1234567</a>
Location: <b>Some street</b>
</pre>",
  "subject": "BookingSG request: Career",
}
`;

exports[`Notification templates tests should create service provider email for citizen updated booking 1`] = `
Object {
  "html": "<pre>
There has been an update to the following booking by the other party.
<br />
Booking for: <b>Career.</b>
<br />
Below is a confirmation of the updated booking details.
<br/>
Booking status: <b>Pending Approval</b>
Date: <b>14 April 2021</b>
Time: <b>10:00am - 11:00am</b>
Video Conference Link: <a href='http://www.zoom.us/1234567'>http://www.zoom.us/1234567</a>
Location: <b>Some street</b>
</pre>",
  "subject": "BookingSG update: Career",
}
`;

exports[`Notification templates tests should create service provider email for service provider cancelled booking 1`] = `
Object {
  "html": "<pre>
You have cancelled the following booking.
<br />
Booking for: <b>Career.</b>
<br />
Booking status: <b>Pending Approval</b>
Date: <b>14 April 2021</b>
Time: <b>10:00am - 11:00am</b>
Video Conference Link: <a href='http://www.zoom.us/1234567'>http://www.zoom.us/1234567</a>
Location: <b>Some street</b>
</pre>",
  "subject": "BookingSG cancellation: Career",
}
`;

exports[`Notification templates tests should create service provider email for service provider updated booking 1`] = `
Object {
  "html": "<pre>
You have updated a booking.
<br />
Booking for: <b>Career.</b>
<br />
Below is a summary of your updated booking details.
<br/>
Booking status: <b>Pending Approval</b>
Date: <b>14 April 2021</b>
Time: <b>10:00am - 11:00am</b>
Video Conference Link: <a href='http://www.zoom.us/1234567'>http://www.zoom.us/1234567</a>
Location: <b>Some street</b>
</pre>",
  "subject": "BookingSG update: Career",
}
`;
